### PR TITLE
User me endpoint cookie rework

### PR DESCRIPTION
**Pull request by feedmypixel**
### This has a breaking change for frontend/Identity - DO NOT merge until [this](https://github.com/guardian/frontend/pull/8042) pull request merged on frontend

I noticed the code for the /user/me endpoint wasn't quite doing what it should be doing and the creation/removal of the GU_MEM cookie wasn't quite right.
This work is involved with the discarding of the GU_MEM cookie and the creation of a minimal GU_MEM cookie for identity signed in users. As well as a good rework of the client side code used to call the /user/me endpoint.

When a user hits the /user/me endpoint:
 - Identity not signed in = 403
 - Identity signed in, not a member = minimal GU_MEM cookie 
 - Identity signed in, member = same response as present 

- Move the creation of the minimal GU_MEM cookie to the backend, this is created when a user has signed in with Identity but does not have a membership tier
- Discard the GU_MEM cookie when a thank you page is shown, covering joining and upgrading,
- Tidy up the client side work involved with creating the minimal GU_MEM cookie
- Tidy up the client side work involved with calling the /user/me endpoint
- Removed client side tests
- Cookie refresh work is no longer needed
- Thnak you page client side work is now being done on the backend

### CODE REVIEW

**Comment by feedmypixel** [comment]
@jamesoram please have a look over this and corresponding front end pull request
@rtyley or @wpf500 can you sanity check the scala work in this pull request

File: frontend/assets/javascripts/src/main.js [hunk 1]
```diff
@@ -84,7 +77,6 @@ require([
     (new JoinFree()).init();
     (new PaidForm()).init();
     (new StaffForm()).init();
-    (new Thankyou()).init(header);
```
Comment (davidrapson) [inline frontend/assets/javascripts/src/main.js]: Nice! Now that `Thankyou` is gone we can also update the Header initialisation to be `(new Header()).init();` as we no longer need to pass the header instance around (which is good)

**Comment by davidrapson** [comment]
Minor comment but really nice cleanup work on the JS :+1:

**Comment by jamesoram** [comment]
:+1:

File: frontend/app/controllers/User.scala [hunk 2]
```diff
@@ -19,8 +19,6 @@ trait User extends Controller {
     Ok(json).withCookies(Cookie("GU_MEM", GuMemCookie.encodeUserJson(json), secure = true, httpOnly = false))
   }
 
-
-
```
Comment (rtyley) [inline frontend/app/controllers/User.scala]: Always worth trying to reduce the size of your diff, so consider leaving out formatting changes if they're not essential (can always tidy up in a separate PR)

File: frontend/app/services/AuthenticationService.scala [hunk 3]
```diff
@@ -20,6 +20,11 @@ trait AuthenticationService {
     user = guUCookieData.user if user.id == minimalSecureUser.id
   } yield IdMinimalUser(user.id, user.publicFields.displayName)
 
+  def identityAuthenticatedUser[A](request: RequestHeader): Option[IdMinimalUser] = for {
```
Comment (rtyley) [inline frontend/app/services/AuthenticationService.scala]: Very curious about this. Why has this been added, when `authenticatedUserFor()` exists above, and does very similar stuff?

Comment (feedmypixel) [inline frontend/app/services/AuthenticationService.scala]: Good spot - I can use the authenticatedUserFor method I will swap this out.

File: frontend/assets/javascripts/src/main.js (hunk 1 above)
Comment (feedmypixel) [inline frontend/assets/javascripts/src/main.js]: thats been added in

**Comment by feedmypixel** [comment]
> Could we maybe add another simpler test for just the decodeBase64 method too? i.e., generic base64 string and test that that is decoded correctly?

tests beefed up

**Comment by feedmypixel** [comment]
[#8042](https://github.com/guardian/frontend/pull/8042) is live on front-end pushing this to membership now

**Comment by feedmypixel** [comment]
on Prod and works with [#8042](https://github.com/guardian/frontend/pull/8042) membership tap on Identity displayed as intended

**Pull request merged**